addons/web/static/src/views/relational_model.js

```diff
@@ -1796,9 +1799,10 @@ export class DynamicRecordList extends DynamicList {
             await this.load();
         }
         await this.model.keepLast.add(this.model.mutex.exec(() => newRecord.load()));
-        this.editedRecord = newRecord;
-        this.onRemoveNewRecord = await this.onCreateRecord(newRecord);
-
+        if (mode === "edit") {
+            this.editedRecord = newRecord;
+            this.onRemoveNewRecord = await this.onCreateRecord(newRecord);
```

Comment:
not sure about this not being called if mode is readonly

Reply:
add a method addExistingRecord